warcraft/contracts/assets/stateful.py

```diff
@@ -0,0 +1,62 @@
+from pyteal import *
+
+def approval_program():
+    app_manager = Bytes("app_manager")
+    total_registered = Bytes("total_registered")
+    escrow = Bytes("escrow")
+    # verify rekey_to and close_rem_to are set as zero_address
+    basic_checks = And(
+        # Always verify that the RekeyTo property of any transaction is set to the ZeroAddress
+        # unless the contract is specifically involved in a rekeying operation.
+        Txn.rekey_to() == Global.zero_address(),
+        Txn.close_remainder_to() == Global.zero_address(),
+        Txn.asset_close_to() == Global.zero_address()
+    )
+
+    on_initialize = Seq([
+        App.globalPut(app_manager, Txn.application_args[0]),
+        App.globalPut(total_registered, Int(0)),
+        Return(Int(1))
```

Comment:
Why do we not initialize escrow.address here (while deploying app)?

Reply:
escrow address changes with application id passed.(because it is hardcoded)

Reply:
It needs app id to confirm that app is being called